src/planner/ConfigurationToEndEffectorOffset.cpp

```diff
@@ -0,0 +1,69 @@
+#include "aikido/planner/ConfigurationToEndEffectorOffset.hpp"
+
+#include "aikido/constraint/Testable.hpp"
+
+namespace aikido {
+namespace planner {
+
+//==============================================================================
+ConfigurationToEndEffectorOffset::ConfigurationToEndEffectorOffset(
+    statespace::ConstStateSpacePtr stateSpace,
+    dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
+    const statespace::StateSpace::State* startState,
+    const Eigen::Vector3d& direction,
+    const double distance,
+    constraint::ConstTestablePtr constraint)
+  : Problem(std::move(stateSpace), std::move(constraint))
+  , mEndEffectorBodyNode(std::move(endEffectorBodyNode))
+  , mStartState(startState)
+  , mDirection(direction.normalized())
+  , mDistance(distance)
+{
+  if (direction.norm() < 1e-6)
```

Comment:
What happens in line 19 if the norm is zero? If it can be successfully normalized, I think we should just allow it.

Reply:
According to the [Eigen documentation](https://eigen.tuxfamily.org/dox/classEigen_1_1MatrixBase.html#ad16303c47ba36f7a41ea264cb26bceb6), zero vector will be unchanged. Otherwise, it'll be successfully normalized. So I just changed this to compare to zero instead of `1e-6`.

If we allow zero vector, then I think we should document this as well.